testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/DeltaQueryExecutor.java

```diff
@@ -0,0 +1,133 @@
+ * limitations under the License.
+ */
+package io.trino.tests.product.utils;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.resolver.ArtifactResolver;
+import io.airlift.resolver.DefaultArtifact;
+import io.trino.tempto.context.TestContext;
+import io.trino.tempto.query.JdbcConnectionsPool;
+import io.trino.tempto.query.JdbcConnectivityParamsState;
+import io.trino.tempto.query.JdbcQueryExecutor;
+import org.sonatype.aether.artifact.Artifact;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.resolver.ArtifactResolver.MAVEN_CENTRAL_URI;
+import static io.airlift.resolver.ArtifactResolver.USER_LOCAL_REPO;
+import static java.lang.ClassLoader.getPlatformClassLoader;
+
+public class DeltaQueryExecutor
+        extends JdbcQueryExecutor
+{
+    private static final DeltaConnectionsPool CONNECTIONS_POOL = new DeltaConnectionsPool();
+
+    public DeltaQueryExecutor(TestContext testContext)
+    {
+        super(testContext.getDependency(JdbcConnectivityParamsState.class, "delta"), CONNECTIONS_POOL, testContext);
+    }
+
+    private static class DeltaConnectionsPool
+            extends JdbcConnectionsPool
+    {
+        private Connection connection;
+
+        @Override
+        public Connection connectionFor(JdbcConnectivityParamsState jdbcParamsState)
+                throws SQLException
+        {
+            if (connection == null || connection.isClosed()) {
+                Properties properties = new Properties();
+                properties.put("user", jdbcParamsState.user);
+                properties.put("password", jdbcParamsState.password);
+                connection = getDatabaseDriver(jdbcParamsState).connect(jdbcParamsState.url, properties);
+            }
+
+            if (connection == null) {
+                // this should never happen, `javax.sql.DataSource#getConnection()` should not return null
+                throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
+            }
+            if (!jdbcParamsState.prepareStatements.isEmpty()) {
+                try (Statement statement = connection.createStatement()) {
+                    for (String query : jdbcParamsState.prepareStatements) {
+                        try {
+                            statement.execute(query);
+                        }
+                        catch (SQLException e) {
+                            throw new SQLException("Preparatory statement failed: " + query, e);
+                        }
+                    }
+                }
+            }
+            return connection;
```

Comment:
once we have a driver, can we delegate to some tempto code instead of copying it here?

Reply:
I think we can't do that for now. The product test module contains two `org.apache.hive.jdbc.HiveDriver` (0.13.1-9 & 3.1.3) after this change. `tempto` may use another driver as the library creates a new connector based on the driver class name.

Reply:
would https://github.com/trinodb/tempto/pull/116 help?